lib/spack/spack/cmd/external.py

```diff
@@ -53,6 +53,9 @@ def setup_parser(subparser):
     find_parser.add_argument(
         "--all", action="store_true", help="search for all packages that Spack knows about"
     )
+    find_parser.add_argument(
+        "--debug-find", action="store_true", help="write all candidate paths seached for externals to stdout"
+    )
```

Comment:
Can we just call it `--debug`? It's an option on `spack external find`, so `spack external find --debug-find` seems redundant. Admittedly `spack -d external find --debug` is possible.